lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs

```diff
@@ -113,8 +114,8 @@ public ICommunicationGroupDriver DefaultGroup
                     {
                         NewCommunicationGroup(_defaultGroupName, _defaultGroupNumberOfTasks);
```

Comment:
This is a problem......DefualtGroup is under lock and within it you are calling NewCommunicationGroup which also tries to grab the lock. The code will just hang.